### PR TITLE
APP-633: Returning HTTP 415 when the payload content-type is not supported.

### PR DESCRIPTION
**Pull request by campidelli**
Payload supports now a MediaType.

### CODE REVIEW

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java [hunk 1]
```diff
@@ -16,66 +16,97 @@
...
+    /**
+     * Holds parameters from one's request.
+     */
+    private Map<String, String> parameters = new HashMap<>();
 
-  /**
-   * Holds headers from one's request.
-   */
-  private Map<String, String> headers = new HashMap<>();
+    /**
```
Comment (crepache) [inline integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java]: Formatter!

**Review by crepache — CHANGES_REQUESTED** [review]

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java [hunk 2]
```diff
@@ -16,66 +16,97 @@
...
+     * Holds headers from one's request.
+     */
+    private Map<String, String> headers = new HashMap<>();
 
-  /**
-   * Payload body.
-   */
-  private String body;
+    /**
```
Comment (crepache) [inline integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java]: Formatter!

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java [hunk 3]
```diff
@@ -16,66 +16,97 @@
...
-   * @param parameters from the request.
-   * @param headers from the request.
-   * @param body from the payload.
-   */
-  public WebHookPayload(Map<String, String> parameters, Map<String, String> headers, String body) {
-    this.parameters = parameters;
-    this.headers = headers;
-    this.body = body;
-  }
+    /**
```
Comment (crepache) [inline integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java]: Formatter!

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java [hunk 4]
```diff
@@ -16,66 +16,97 @@
...
+    }
 
-  /**
-   * Returns the payload parameters.
-   * @return request parameters.
-   */
-  public Map<String, String> getParameters() {
-    return parameters;
-  }
+    /**
```
Comment (crepache) [inline integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java]: Formatter!

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java [hunk 5]
```diff
@@ -16,66 +16,97 @@
...
+    }
 
-  /**
-   * Returns the payload body.
-   * @return the payload body.
-   */
-  public String getBody() {
-    return body;
-  }
+    /**
```
Comment (crepache) [inline integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java]: Formatter!

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookIntegration.java [hunk 6]
```diff
@@ -536,4 +538,28 @@ protected IntegrationInstance getIntegrationInstance(String instanceId) {
     return application.getWhiteList();
   }
 
+    /**
```
Comment (crepache) [inline integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookIntegration.java]: Formatter!

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java [hunk 7]
```diff
@@ -16,66 +16,97 @@
...
+    }
 
-  /**
-   * Returns the request headers.
-   * @return request headers.
-   */
-  public Map<String, String> getHeaders() {
-    return headers;
-  }
+    /**
```
Comment (crepache) [inline integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java]: Formatter!

**Review by crepache — CHANGES_REQUESTED** [review]

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java (hunk 1 above)
Comment (campidelli) [reply integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java]: Ok, done!

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java (hunk 2 above)
Comment (campidelli) [reply integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java]: Ok, done!

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java (hunk 3 above)
Comment (campidelli) [reply integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java]: Ok, done!

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java (hunk 4 above)
Comment (campidelli) [reply integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java]: Ok, done!

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java (hunk 5 above)
Comment (campidelli) [reply integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java]: Ok, done!

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java (hunk 7 above)
Comment (campidelli) [reply integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookPayload.java]: Ok, done!

File: integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookIntegration.java (hunk 6 above)
Comment (campidelli) [reply integration-webhook/src/main/java/org/symphonyoss/integration/webhook/WebHookIntegration.java]: Ok, done!

**Review by crepache — APPROVED** [review]

**Pull request merged**